pkg/controller/tenant/tenant_controller.go

```diff
@@ -180,24 +181,84 @@ func (tc *tenantController) syncTenant(tenantName string) (err error) {
 		klog.V(4).Infof("Finished initializing tenant %q (%v)", tenantName, time.Since(startTime))
 	}()
 
-	createFailures := []error{}
+	if err, done := tc.createNamespaces(tenantName); !done {
+		return err
+	}
 
-	// Create Default namespaces
-	for _, nsName := range tenant_default_namespaces {
+	if err, done := tc.createInitialRoleAndBinding(tenantName); !done {
+		return err
+	}
+
+	return nil
+}
 
+func (tc *tenantController) createNamespaces(tenantName string) (error, bool) {
+	failures := []error{}
+	for _, nsName := range tenantDefaultNamespaces {
 		ns := v1.Namespace{
 			ObjectMeta: metav1.ObjectMeta{Tenant: tenantName, Name: nsName},
 		}
 		if _, err := tc.kubeClient.CoreV1().NamespacesWithMultiTenancy(tenantName).Create(&ns); err != nil && !errors.IsAlreadyExists(err) {
-			createFailures = append(createFailures, err)
+			failures = append(failures, err)
 		}
 	}
-
-	if len(createFailures) != 0 {
-		ret := utilerrors.Flatten(utilerrors.NewAggregate(createFailures))
+	if len(failures) != 0 {
+		ret := utilerrors.Flatten(utilerrors.NewAggregate(failures))
 		klog.Errorf("Errors happened in tenant initialization of %v: %v", tenantName, ret)
-		return ret
+		return ret, false
 	}
+	return nil, true
+}
 
-	return nil
+func (tc *tenantController) createInitialRoleAndBinding(tenant string) (error, bool) {
+	// tenant admin act as cluster admin for tha tenant
+	roleName := initialClusterRoleName(tenant)
+	role := &rbacv1.ClusterRole{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:   roleName,
+			Tenant: tenant,
+		},
+		Rules: []rbacv1.PolicyRule{initialClusterRoleRules()},
+	}
+
+	binding := &rbacv1.ClusterRoleBinding{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:   initialClusterRoleBindingName(tenant),
+			Tenant: tenant,
+		},
+		Subjects: []rbacv1.Subject{
+			{Kind: rbacv1.UserKind, Name: "admin"},
+		},
+		RoleRef: rbacv1.RoleRef{APIGroup: rbacv1.GroupName, Kind: "ClusterRole", Name: initialClusterRoleName(tenant)},
+	}
+
+	var failures []error
+	if _, err := tc.kubeClient.RbacV1().ClusterRolesWithMultiTenancy(tenant).Create(role); err != nil && !errors.IsAlreadyExists(err) {
+		failures = append(failures, err)
+	}
+	if _, err := tc.kubeClient.RbacV1().ClusterRoleBindingsWithMultiTenancy(tenant).Create(binding); err != nil && !errors.IsAlreadyExists(err) {
+		failures = append(failures, err)
+	}
+	if len(failures) != 0 {
+		ret := utilerrors.Flatten(utilerrors.NewAggregate(failures))
+		klog.Errorf("Errors happened when creating initial cluster role and binding for tenant %s: %v", tenant, ret)
+		return ret, false
+	}
+	return nil, true
+}
+
+func initialClusterRoleName(tenant string) string {
+	return tenant + "-admin-role"
+}
+
+func initialClusterRoleBindingName(tenant string) string {
+	return initialClusterRoleName(tenant) + "-binding"
+}
+
+func initialClusterRoleRules() rbacv1.PolicyRule {
+	return rbacv1.PolicyRule{
+		Verbs:     []string{"*"},
+		APIGroups: []string{"*"},
+		Resources: []string{"*"},
+	}
```

Comment:
We should exclude the resources, such as daemonset/nodes, which are accessible for cluster admin only.

Reply:
Maybe we can leave it as it for now, as it would need to another PR to discover all those resources and put them here.

But we should log an issue and put some comment here.

Reply:
yeah I've been back and forth about this. the idea is for resource like daemonset/node, the attempt to access will be blocked by other admission mechanism. we have [an issue](https://github.com/futurewei-cloud/arktos/issues/39) to track this.

Reply:
+1 on this. Other mechanisms would block that.  Also in some cases we might even allow  creating nodes in a tenant space.  So it's better not to assume it in the default admin role settings.